tools/fapi/tss2_template.c

```diff
@@ -250,6 +250,75 @@ TSS2_RC auth_callback(
     return TSS2_RC_SUCCESS;
 }
 
+TSS2_RC sign_callback(
+    char        const *description,
+    char        const *objectPath,
+    char        const *publicKey,
+    char        const *publicKeyHint,
+    uint32_t    hashAlg,
+    uint8_t     const *dataToSign,
+    size_t      dataToSignSize,
+    uint8_t     const **signature,
+    size_t      *signatureSize,
+    __attribute__((unused)) void *userData)
+{
+    /* The public key that will be used by the TPM to verify the signature in PEM encoding */
+    (void) publicKey; // 
+    (void) publicKeyHint;
+    (void) hashAlg;
+    (void) dataToSign;
+    (void) dataToSignSize;
```

Comment:
these are all used AFAICT.

Reply:
In the current version almost all parameters are used now. The user gets a message in this form: 
```
PolicySigned: Authorize usage of P_RSA2048SHA256/HS/SRK/mySignKey by signing the nonce with the key corresponding to "My Signature Key" an the hash algorithm "sha1"
```

```
description=PolicySigned
objectPath=P_RSA2048SHA256/HS/SRK/mySignKey
publicKeyHint=My Signature Key (optional)
tpm2_alg_util_algtostr(hashAlg)=sha1
```

However, we are currently thinking about how to display the parameter "publicKey" (PEM-encoded public key). In my opinion it is too long for stdout. It could be written to a separate file if the user requires this. This could be implemented as separate parameter or with another user prompt...

Reply:
Oh yeah, that could be tricky, especially if the user doesn't have the hint set up. Perhaps if hint isn't specified, couldn't you do a fingerprint of the public key? Maybe something like  Base64 encoded SHA-256? I am not sure if their is a standard anywhere for these. I kinda perused some interwebs things like:
https://superuser.com/questions/421997/what-is-a-ssh-key-fingerprint-and-how-is-it-generated

Reply:
Ok. I implemented these two variants depending if the key hint is present:
```
PolicySigned: Authorize usage of P_RSA2048SHA256/HS/SRK/mySignKey1 by signing the nonce with the key corresponding to the fingerprint "SHA256:Gf8QGWT2MOy8tOfpe6m+6MUUm2GUijGPkvCTjtJPOJz//XMom+k+7OaWmA==" and the hash algorithm "sha1"
```

```
PolicySigned: Authorize usage of P_RSA2048SHA256/HS/SRK/mySignKey2 by signing the nonce with the key corresponding to the key hint "My Signature Key" an the hash algorithm "sha1"
```